llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -4229,29 +4207,18 @@ bool AArch64FrameLowering::assignCalleeSavedSpillSlots(
 
   // Insert VG into the list of CSRs, immediately before LR if saved.
   if (requiresSaveVG(MF)) {
-    std::vector<CalleeSavedInfo> VGSaves;
-    SMEAttrs Attrs = AFI->getSMEFnAttrs();
-
-    auto VGInfo = CalleeSavedInfo(AArch64::VG);
-    VGInfo.setRestored(false);
-    VGSaves.push_back(VGInfo);
-
-    // Add VG again if the function is locally-streaming, as we will spill two
-    // values.
-    if (Attrs.hasStreamingBody() && !Attrs.hasStreamingInterface())
-      VGSaves.push_back(VGInfo);
-
-    bool InsertBeforeLR = false;
+    CalleeSavedInfo VGInfo(AArch64::VG);
 
+    bool InsertedBeforeLR = false;
     for (unsigned I = 0; I < CSI.size(); I++)
       if (CSI[I].getReg() == AArch64::LR) {
-        InsertBeforeLR = true;
-        CSI.insert(CSI.begin() + I, VGSaves.begin(), VGSaves.end());
+        InsertedBeforeLR = true;
```

Comment:
unrelated to your patch, but this could be a simple:
```
auto It = find_if(
    CSI, [](CalleeSavedInfo &CSR) { return CSR.getReg() == AArch64::LR; });
if (It != CSI.end())
  CSI.insert(It, VGInfo);
else
  CSI.push_back(VGInfo);
```
which I think would be easier to read.